File: vehicle.dart
// Define the class
class Vehicle{
  double price;
  String color;
  String model;
  
  //Constructor
  Vehicle(double price,String color,String model){
    this.price = price;
    this.color = color;
    this.model = model;
  }
  
  // Method for printing details
  void printVehicleDetails(){
    print("Price : ${this.price}");
    print("Color: " + this.color);
    print("Model: " + this.model);
  }
}

void main() {
  Vehicle car1 = new Vehicle(500, "Red" ,"008x5");

  car1.printVehicleDetails();
}

File: vehicle.ts
class Vehicle {
  price: number;
  color: string;
  model: string;

  constructor(thePrice: number, theColor: string, theModel: string) {
    this.price = thePrice;
    this.color = theColor;
    this.model = theModel;
  }

  showDetails() {
    console.log(`Price: ${this.price}`);
    console.log(`Color: ${this.color}`);
    console.log(`Model: ${this.model}`);
  }
}

let carOne = new Vehicle(500, 'Red', '008x5');
carOne.showDetails();
